app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -587,22 +587,34 @@ def validate(self, data):
             "status": NimbusConstants.STATUS_ALLOWS_UPDATE,
             "publish_status": NimbusConstants.PUBLISH_STATUS_ALLOWS_UPDATE,
         }
+        update_exempt_fields = NimbusExperiment.STATUS_UPDATE_EXEMPT_FIELDS
 
         if self.instance:
             for status_field, restricted_statuses in restrictive_statuses.items():
                 current_status = getattr(self.instance, status_field)
-                is_locked = current_status not in restricted_statuses
-                modifying_fields = set(data.keys()) - set(
-                    NimbusExperiment.STATUS_UPDATE_EXEMPT_FIELDS
+
+                restricted_stats = restricted_statuses["all"] + (
+                    (restricted_statuses["rollouts"])
+                    if self.instance.is_rollout
+                    else restricted_statuses["experiments"]
+                )
+                exempt_fields = update_exempt_fields["all"] + (
+                    (update_exempt_fields["rollouts"])
+                    if self.instance.is_rollout
+                    else update_exempt_fields["experiments"]
```

Comment:
oh we can avoid 2 ifs, if we check first `if self.instance.is_rollout` and set restricted status and exempt fields for eg
```python
if self.instance.is_rollout:
    restricted_status = "rollout key"
else
   restricted_status = "experiment key"
```

Reply:
Hopefully the updated version is clearer. Let me know what you think